src/cryptonoteprotocol/CryptoNoteProtocolHandler.cpp

```diff
@@ -334,11 +335,15 @@ namespace CryptoNote
 
 	    uint64_t days;
 
-	    if (currentHeight >= CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT || remoteHeight < CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT)
+	    if (currentHeight >= CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT || remoteHeight < CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT)
```

Comment:
This block doesn't seem to work as intended. The intent is to figure out the amount of days behind the daemon we're syncing with. This causes a few issues when we change difficulty, since we use the average block time to calculate the total days.

If we are at block 0, and the remote daemon is above the v3 height, then we have 3 calculations to perform:
Days from block 0 to DIFFICULTY_TARGET_V2_HEIGHT - This will be DIFFICULTY_TARGET_V2 (10 seconds) times DIFFICULTY_TARGET_V2_HEIGHT (not sure what that is..) but you'll get the estimated time for that many blocks to have elapsed.

But then we also need to add on the estimated time for the blocks with the DIFFICULTY_TARGET_V2 time per block, and the V3 one.

I think we could maybe simplify this whole thing.

```cpp
uint64_t blocksWithDiffV3Remaining = 0;
uint64_t blocksWithDiffV2Remaining = 0;

/* We have passed V3 height. Lets figure out how many v3 diff blocks we still need to sync. */
if (remoteHeight >= CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT)
{
	/* Take the max of the current height or the target height. If our current height is less than the diff v3 target height,
	 * we don't want to include those blocks, as they are not v3 ones. */
	blocksWithDiffV3Remaining = remoteHeight - std::max(currentHeight, CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT);
}

/* We have passed V3 height. Lets figure out how many v3 diff blocks we still need to sync. */
if (remoteHeight >= CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT)
{
	/* Find number of blocks between max(current height / v2 height) and v3 height */
	blocksWithDiffV2Remaining = remoteHeight - blocksWithDiffV3Remaining - std::max(currentHeight, CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT);
}

const uint64_t blocksWithDiffV1Remaining = remoteHeight - blocksWithDiffV3Remaining - blocksWithDiffV2Remaining - currentHeight;

const uint64_t days = blocksWithDiffV3Remaining / (24 * 60 * 60 / CryptoNote::parameters::DIFFICULTY_TARGET_V3);
days += blocksWithDiffV2Remaining / (24 * 60 * 60 / CryptoNote::parameters::DIFFICULTY_TARGET_V2);
days += blocksWithDiffV1Remaining / (24 * 60 * 60 / CryptoNote::parameters::DIFFICULTY_TARGET_V1);
```

Reply:
love the simplification! 
just had to change the const uint64_t days = blocksWithDiffV3Remaining to days = blocksWithDiffV3Remaining to avoid declaring as read-only constant which throws error while building in the following two lines?

Reply:
yep shouldn't be const